sunpy/visualization/animator/tests/test_wcs.py

```diff
@@ -0,0 +1,155 @@
+    CDELT1  =                  0.4 / [min] Coordinate increment at reference point
+    CDELT2  =                2E-11 / [m] Coordinate increment at reference point
+    CDELT3  =   0.0027777777777778 / [deg] Coordinate increment at reference point
+    CDELT4  =   0.0013888888888889 / [deg] Coordinate increment at reference point
+    CUNIT1  = 'min'                / Units of coordinate increment and value
+    CUNIT2  = 'm'                  / Units of coordinate increment and value
+    CUNIT3  = 'deg'                / Units of coordinate increment and value
+    CUNIT4  = 'deg'                / Units of coordinate increment and value
+    CTYPE1  = 'TIME'               / Coordinate type code
+    CTYPE2  = 'WAVE'               / Vacuum wavelength (linear)
+    CTYPE3  = 'HPLT-TAN'           / Coordinate type codegnomonic projection
+    CTYPE4  = 'HPLN-TAN'           / Coordinate type codegnomonic projection
+    CRVAL1  =                  0.0 / [min] Coordinate value at reference point
+    CRVAL2  =                  0.0 / [m] Coordinate value at reference point
+    CRVAL3  =                  0.0 / [deg] Coordinate value at reference point
+    CRVAL4  =                  0.0 / [deg] Coordinate value at reference point
+    LONPOLE =                180.0 / [deg] Native longitude of celestial pole
+    LATPOLE =                  0.0 / [deg] Native latitude of celestial pole
+    """)
+    return WCS(header=header)
+
+
+@pytest.mark.parametrize("data, slices, dim", (
+    (np.arange(120).reshape((5, 4, 3, 2)), [0, 0, 'x', 'y'], 2),
+    (np.arange(120).reshape((5, 4, 3, 2)), [0, 'x', 0, 'y'], 2),
+    (np.arange(120).reshape((5, 4, 3, 2)), ['x', 0, 0, 'y'], 2),
+    (np.arange(120).reshape((5, 4, 3, 2)), ['y', 0, 'x', 0], 2),
+    (np.arange(120).reshape((5, 4, 3, 2)), [0, 0, 0, 'x'], 1),
+    ))
+def test_construct_array_animator(wcs_4d, data, slices, dim):
+    if dim == 1:
+        pytest.importorskip("astropy", minversion="4.0dev26173")
+
+    array_animator = ArrayAnimatorWCS(data, wcs_4d, slices)
+
+    assert isinstance(array_animator, ArrayAnimatorWCS)
+    assert array_animator.plot_dimensionality == dim
+    assert array_animator.num_sliders == data.ndim - dim
+    for i, (wslice, arange) in enumerate(zip(slices, array_animator.axis_ranges[::-1])):
+        if wslice not in ['x', 'y']:
+            assert callable(arange)
+            a = arange(0)
+            if "pos" in wcs_4d.world_axis_physical_types[i]:
+                assert not isinstance(a, u.Quantity)
+                assert a is 0
+            else:
+                assert isinstance(a, u.Quantity)
+                assert a.value == wcs_4d.pixel_to_world_values(*[0]*wcs_4d.world_n_dim)[i]
+                assert a.unit == wcs_4d.world_axis_units[i]
+        else:
+            assert arange is None
+
+
+def test_constructor_errors(wcs_4d):
+    # WCS is not BaseLowLevelWCS
+    with pytest.raises(ValueError):
```

Comment:
Could you add the `match='string that is in error message'` bit to `pytest.raises` here, so it's easier to read exactly what is being caught.